File: backend/src/main/java/com/spinwish/backend/models/responses/users/DJRegisterResponse.java
package com.spinwish.backend.models.responses.users;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "DJ registration response with DJ profile information")
public class DJRegisterResponse {
    
    @Schema(description = "User's email address", example = "john.smith@example.com")
    private String emailAddress;
    
    @Schema(description = "User's full name", example = "John Smith")
    private String username;
    
    @Schema(description = "DJ's stage name", example = "DJ Nexus")
    private String djName;
    
    @Schema(description = "DJ's biography", example = "Electronic music producer with 8+ years of experience...")
    private String bio;
    
    @Schema(description = "List of music genres", example = "[\"House\", \"Techno\", \"Electronic\"]")
    private List<String> genres;
    
    @Schema(description = "DJ's Instagram handle", example = "@dj_nexus_official")
    private String instagramHandle;
    
    @Schema(description = "DJ's profile image URL", example = "https://example.com/profile.jpg")
    private String profileImage;
    
    @Schema(description = "Initial rating (0.0 for new DJs)", example = "0.0")
    private Double rating;
    
    @Schema(description = "Initial followers count", example = "0")
    private Integer followers;
    
    @Schema(description = "Account verification status", example = "false")
    private Boolean emailVerified;
    
    @Schema(description = "Success message", example = "DJ registration successful. Please check your email for verification.")
    private String message;
}

File: spinwishapp/lib/models/auth/dj_register_response.dart
class DJRegisterResponse {
  final String emailAddress;
  final String username;
  final String djName;
  final String bio;
  final List<String> genres;
  final String? instagramHandle;
  final String? profileImage;
  final double rating;
  final int followers;
  final bool emailVerified;
  final String message;

  DJRegisterResponse({
    required this.emailAddress,
    required this.username,
    required this.djName,
    required this.bio,
    required this.genres,
    this.instagramHandle,
    this.profileImage,
    required this.rating,
    required this.followers,
    required this.emailVerified,
    required this.message,
  });

  factory DJRegisterResponse.fromJson(Map<String, dynamic> json) => DJRegisterResponse(
    emailAddress: json['emailAddress'] ?? '',
    username: json['username'] ?? '',
    djName: json['djName'] ?? '',
    bio: json['bio'] ?? '',
    genres: List<String>.from(json['genres'] ?? []),
    instagramHandle: json['instagramHandle'],
    profileImage: json['profileImage'],
    rating: (json['rating'] ?? 0.0).toDouble(),
    followers: json['followers'] ?? 0,
    emailVerified: json['emailVerified'] ?? false,
    message: json['message'] ?? '',
  );

  Map<String, dynamic> toJson() => {
    'emailAddress': emailAddress,
    'username': username,
    'djName': djName,
    'bio': bio,
    'genres': genres,
    'instagramHandle': instagramHandle,
    'profileImage': profileImage,
    'rating': rating,
    'followers': followers,
    'emailVerified': emailVerified,
    'message': message,
  };
}
